ui/ducks/send/swap-and-send-utils.ts

```diff
@@ -0,0 +1,198 @@
+};
+
+type Quote = {
+  gasParams: {
+    maxGas: number;
+  };
+  trade: {
+    data: string;
+    to: string;
+    from: string;
+    value: string;
+  };
+  approvalNeeded: null | {
+    data: string;
+    to: string;
+    from: string;
+  };
+  sourceAmount: string;
+  destinationAmount: string;
+  sourceToken: string;
+  destinationToken: string;
+  sender: string;
+  recipient: string;
+  aggregator: string;
+  aggregatorType: string;
+  error: null | object;
+  fee: number;
+};
+
+const QUOTE_VALIDATORS = [
+  {
+    property: 'gasParams',
+    type: 'object',
+    validator: (gasParams: Record<string, any>) =>
+      gasParams && isNumber(gasParams.maxGas),
+  },
+  {
+    property: 'trade',
+    type: 'object',
+    validator: (trade: Record<string, any>) =>
+      trade &&
+      validHex(trade.data) &&
+      isValidHexAddress(trade.to, { allowNonPrefixed: false }) &&
+      isValidHexAddress(trade.from, { allowNonPrefixed: false }) &&
+      truthyString(trade.value),
+  },
+  {
+    property: 'approvalNeeded',
+    type: 'object',
+    validator: (approvalTx: Record<string, any>) =>
+      approvalTx === null ||
+      (approvalTx &&
+        validHex(approvalTx.data) &&
+        isValidHexAddress(approvalTx.to, { allowNonPrefixed: false }) &&
+        isValidHexAddress(approvalTx.from, { allowNonPrefixed: false })),
+  },
+  {
+    property: 'sourceAmount',
+    type: 'string',
+    validator: truthyDigitString,
+  },
+  {
+    property: 'destinationAmount',
+    type: 'string',
+    validator: truthyDigitString,
+  },
+  {
+    property: 'sourceToken',
+    type: 'string',
+    validator: (input: string) =>
+      isValidHexAddress(input, { allowNonPrefixed: false }),
+  },
+  {
+    property: 'destinationToken',
+    type: 'string',
+    validator: (input: string) =>
+      isValidHexAddress(input, { allowNonPrefixed: false }),
+  },
+  {
+    property: 'sender',
+    type: 'string',
+    validator: (input: string) =>
+      isValidHexAddress(input, { allowNonPrefixed: false }),
+  },
+  {
+    property: 'recipient',
+    type: 'string',
+    validator: (input: string) =>
+      isValidHexAddress(input, { allowNonPrefixed: false }),
+  },
+  {
+    property: 'aggregator',
+    type: 'string',
+    validator: truthyString,
+  },
+  {
+    property: 'aggregatorType',
+    type: 'string',
+    validator: truthyString,
+  },
+  {
+    property: 'error',
+    type: 'object',
+    validator: (error: any) => error === null || typeof error === 'object',
+  },
+  {
+    property: 'fee',
+    type: 'number',
+  },
+];
+const SWAPS_API_VERSION = 'v2';
+
+export async function getSwapAndSendQuotes(
+  request: Request,
+): Promise<Record<string, Quote>> {
+  const { chainId, ...params } = request;
+
+  const queryString = new URLSearchParams(params);
+
+  // FIXME: can't use the dev API since it's several major versions behind prod
+  const url = `${SWAPS_API_V2_BASE_URL}/${SWAPS_API_VERSION}/networks/${chainId}/quotes?${queryString}`;
+
```

Comment:
I noticed we use a different endpoint than swaps. Do you know the difference between quotes and trades? Also, are they using V2?

Reply:
It was added in this PR: https://github.com/consensys-vertical-apps/va-mmcx-swap-api/pull/26

Not sure about the differences besides quotes having the `recipient` param we need for swap and send.

Reply:
Asked Callum and he confirmed there are breaking changes between the versions. We can inspect the types by doing this:

"Yes, trades and quotes and breaking changes - you can see the API documentation by running the API in dev mode (see readme) and we plan to eventually deprecate trades and move to v2 over time"

Reply:
Yeah the typing lines up